configure.py

```diff
@@ -394,7 +394,7 @@ def binary(name):
 # Search for generated headers relative to build dir.
```

Comment:
If we're going to move includes around like this, i personally have a strong preference for keeping public headers separated entirely from the internal headers.

ninja.git/
  - src/
    - *.cc
    - *.h
  - inc/ninja
    - *.h

This allows for package managers to simply "cp -r inc/ /usr/inc/" in order to install the public API.